extensions/schema-compare/src/schemaCompareResult.ts

```diff
@@ -0,0 +1,361 @@
+			this.flexModel.addItem(this.splitView);
+
+			// only enable generate script button if the target is a db
+			if (this.targetEndpointInfo.endpointType === azdata.SchemaCompareEndpointType.database) {
+				this.generateScriptButton.enabled = true;
+			} else {
+				this.generateScriptButton.title = localize('schemaCompare.generateScriptButtonDisabledTitle', 'Generate script is enabled when the target is a database');
+			}
+		} else {
+			this.flexModel.addItem(this.noDifferencesLabel, { CSSStyles: { 'margin': 'auto' } });
+		}
+
+		let sourceText = '';
+		let targetText = '';
+		this.differencesTable.onRowSelected(() => {
+			let difference = this.comparisonResult.differences[this.differencesTable.selectedRows[0]];
+			if (difference !== undefined) {
+				sourceText = difference.sourceScript === null ? '\n' : this.getAggregatedScript(difference, true);
+				targetText = difference.targetScript === null ? '\n' : this.getAggregatedScript(difference, false);
+
+				this.diffEditor.updateProperties({
+					contentLeft: sourceText,
+					contentRight: targetText,
+					title: localize('schemaCompare.ObjectDefinitionsTitle', 'Object Definitions')
+				});
+			}
+		});
+	}
+
+	private getAllDifferences(differences: azdata.DiffEntry[]): string[][] {
+		let data = [];
+		if (differences) {
+			differences.forEach(difference => {
+				if (difference.differenceType === azdata.SchemaDifferenceType.Object) {
+					if (difference.sourceValue !== null || difference.targetValue !== null) {
+						data.push([difference.name, difference.sourceValue, this.SchemaCompareActionMap[difference.updateAction], difference.targetValue]);
+					}
+				}
+			});
+		}
+
+		return data;
+	}
+
+	private getAggregatedScript(diffEntry: azdata.DiffEntry, getSourceScript: boolean): string {
+		let script = '';
+		if (diffEntry !== null) {
+			script += getSourceScript ? diffEntry.sourceScript : diffEntry.targetScript;
+			diffEntry.children.forEach(child => {
+				let childScript = this.getAggregatedScript(child, getSourceScript);
+				if (childScript !== 'null') {
+					script += childScript;
+				}
+			});
+		}
+		return script;
+	}
+
+	private reExecute(): void {
+		this.flexModel.removeItem(this.splitView);
+		this.flexModel.removeItem(this.noDifferencesLabel);
+		this.flexModel.addItem(this.loader, { CSSStyles: { 'margin-top': '30px' } });
+		this.diffEditor.updateProperties({
+			contentLeft: '\n',
+			contentRight: '\n'
+		});
+		this.differencesTable.selectedRows = null;
+		this.resetButtons();
+		this.execute();
+	}
+
+	private createCompareButton(view: azdata.ModelView): void {
+		let runIcon = path.join(__dirname, '.', 'media', 'compare.svg');
+
+		this.compareButton = view.modelBuilder.button().withProperties({
+			label: localize('schemaCompare.compareButton', 'Compare'),
+			iconPath: runIcon,
+			title: localize('schemaCompare.compareButtonTitle', 'Compare')
+		}).component();
+
+		this.compareButton.onDidClick(async (click) => {
```

Comment:
event handlers are all disposable, you should dispose them when the editor window is getting closed.

Reply:
@alanrenmsft looks like the ModelViewEditorImpl uses IEditorService which doesn't expose a closeEditor. This might be a bigger change - would you mind if I investigate this more and take this in separate PR or do you think its blocking?

Ref : https://github.com/Microsoft/azuredatastudio/blob/15a19c044db9a753361f83e25bd38ef4e959de1e/src/sql/workbench/api/node/mainThreadModelViewDialog.ts#L58

https://github.com/Microsoft/azuredatastudio/blob/15a19c044db9a753361f83e25bd38ef4e959de1e/src/vs/workbench/services/editor/common/editorService.ts#L123